### PR TITLE
Adding the ISOdate functionality to the `NexusWellMod` class.

### PR DESCRIPTION
**Pull request by cflynn3**

### CODE REVIEW

**Review by kwr3 — APPROVED** [review]
Looks good

File: ResSimpy/Nexus/DataModels/NexusWellMod.py [hunk 1]
```diff
@@ -68,3 +77,7 @@ def get_keyword_mapping() -> dict[str, tuple[str, type]]:
             'KHMULT': ('perm_thickness_mult', float),
         }
         return keyword_mapping
+
+    @property
+    def units(self) -> BaseUnitMapping:
+        raise NotImplementedError("Units not implemented for WellMods yet")  # TODO: Implement units for this class.
```
Comment (jonnyh404) [inline ResSimpy/Nexus/DataModels/NexusWellMod.py:83]: I think this should just return a CompletionUnits instance.

Comment (jonnyh404) [reply ResSimpy/Nexus/DataModels/NexusWellMod.py:83]:
```
@property
def units(self) -> CompletionUnits:
     return CompletionUnits(self.unit_system)
```

**Review by jonnyh404 — APPROVED** [review]
Looks good to me. Please change units to CompletionUnit

**Pull request merged**